sdk/core/azure-core/src/test/java/com/azure/core/models/CloudEventTests.java

```diff
@@ -0,0 +1,309 @@
+            + "  \"time\": \"2020-07-21T18:41:31.166Z\",\n"
+            + "  \"specversion\": \"1.0\"\n"
+            + "}";
+        assertThrows(IllegalArgumentException.class, () -> {
+            CloudEvent.fromString(cloudEventJson);
+        });
+    }
+
+    @Test
+    public void consumeCloudEventWithoutArrayBrackets() throws IOException {
+        String jsonData = getTestPayloadFromFile("CloudEventNoArray.json");
+
+        List<CloudEvent> events = CloudEvent.fromString(jsonData);
+
+        assertNotNull(events);
+        assertEquals(1, events.size());
+
+        ContosoItemReceivedEventData data = events.get(0).getData().toObject(ContosoItemReceivedEventData.class, SERIALIZER);
+        assertNotNull(data);
+
+        assertEquals("512d38b6-c7b8-40c8-89fe-f46f9e9622b6", data.getItemSku());
+
+        Map<String, Object> additionalProperties = events.get(0).getExtensionAttributes();
+
+        assertNotNull(additionalProperties);
+        assertTrue(additionalProperties.containsKey("foo"));
+        assertEquals("bar", additionalProperties.get("foo"));
+    }
+
+    @Test
+    public void consumeCloudEventWithNullData() throws IOException {
+        String jsonData = getTestPayloadFromFile("CloudEventNullData.json");
+
+        List<CloudEvent> events = CloudEvent.fromString(jsonData);
+
+        assertNotNull(events);
+        assertEquals(1, events.size());
+
+        assertNull(events.get(0).getData());
+    }
+
+    @Test
+    public void consumeCloudEventWithBinaryData() throws IOException {
+        String jsonData = getTestPayloadFromFile("CloudEventBinaryData.json");
+
+        byte[] data = "c2FtcGxlYmluYXJ5ZGF0YQ==".getBytes(StandardCharsets.UTF_8);
+
+        List<CloudEvent> events = CloudEvent.fromString(jsonData);
+
+        assertNotNull(events);
+        assertEquals(1, events.size());
+
+        byte[] eventData = events.get(0).getData().toBytes();
+
+        assertNotNull(eventData);
+
+        assertArrayEquals(data, eventData);
+    }
+
+    @Test
+    public void consumeCloudEventJsonData() throws IOException {
+        String jsonData = getTestPayloadFromFile("CloudEventJsonData.json");
+
+        List<CloudEvent> events = CloudEvent.fromString(jsonData);
+
+        assertNotNull(events);
+        assertEquals(1, events.size());
+
+        ContosoItemReceivedEventData data = events.get(0).getData().toObject(ContosoItemReceivedEventData.class,
+            SERIALIZER);
+        assertNotNull(data);
+
+        assertEquals("512d38b6-c7b8-40c8-89fe-f46f9e9622b6", data.getItemSku());
+
+        Map<String, Object> additionalProperties = events.get(0).getExtensionAttributes();
+
+        assertNotNull(additionalProperties);
+        assertTrue(additionalProperties.containsKey("foo"));
+        assertEquals("bar", additionalProperties.get("foo"));
+    }
+
+    @Test
+    public void consumeCloudEventXmlData() throws IOException {
+        String jsonData = getTestPayloadFromFile("CloudEventXmlData.json");
```

Comment:
String jsonData -> String xmlData

Reply:
It's indeed json data because this variable is for the CloudEvent. The data is xml. Since this name is misleading, I changed it to  cloudEventJson